### PR TITLE
Automated cherry pick of #26792 upstream release 1.2

### PR DESCRIPTION
**Pull request by janetkuo**
Automated cherry pick of #26792 upstream release 1.2
Ref  #26724 

@pwittrock @bgrant0607 @ihmccreery 

[Analytics]()

### CODE REVIEW

**Comment by janetkuo** [comment]
test flakes against non-HEAD 
@k8s-bot unit test this issue: #IGNORE

**Comment by janetkuo** [comment]
@k8s-bot unit test this issue: #IGNORE

File: pkg/util/deployment/deployment.go [hunk 1]
```diff
@@ -111,11 +111,34 @@ func ListPods(deployment *extensions.Deployment, getPodList podListFunc) (*api.P
...
 
+// equalIgnoreHash returns true if two given podTemplateSpec are equal, ignoring the diff in value of Labels[pod-template-hash]
+// We ignore pod-template-hash because the hash result would be different upon podTemplateSpec API changes
+// (e.g. the addition of a new field will cause the hash code to change)
+// Note that we assume input podTemplateSpecs contain non-empty labels
+func equalIgnoreHash(template1, template2 api.PodTemplateSpec) (bool, error) {
+	// The podTemplateSpec must have a non-empty label so that label selectors can find them.
+	// This is checked by validation (of resources contain a podTemplateSpec).
+	if len(template1.Labels) == 0 || len(template2.Labels) == 0 {
+		return false, fmt.Errorf("Unexpected empty labels found in given template")
```
Comment (pwittrock) [inline pkg/util/deployment/deployment.go]: Out of scope for this PR, but for failed sanity checks such as this one, we may want to start putting stack traces in the error to help debug what is going on.

**Comment by janetkuo** [comment]
@roberthbailey can we merge this?

**Comment by ihmccreery** [comment]
@roberthbailey Please merge this asap tomorrow, so we can get results over the weekend.

**Comment by roberthbailey** [comment]
github says the PR is out of date with the base branch.

**Comment by janetkuo** [comment]
> github says the PR is out of date with the base branch.

I'm not sure what that means? Do I need to rebase? Or should I click the "Update branch" button?
screenshot from 2016-06-10 09 59 24

**Comment by roberthbailey** [comment]
Yes, it means you need to rebase. You can either do it yourself or click the button. See https://github.com/blog/2051-protected-branches-and-required-status-checks

**Comment by janetkuo** [comment]
Clicked the "update branch" button and re-applied LGTM.

**Comment by ihmccreery** [comment]
Taking LGTM off; please squash the merge commit.  See https://github.com/kubernetes/kubernetes/blob/5b5c8076ccbc4babc7af6c7619a16d5a80b44ec2/docs/devel/faster_reviews.md#7-fix-feedback-in-a-new-commit and/or https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits.  I can help you on Monday if anything doesn't go well.

**Pull request merged**